.clang-tidy

```diff
@@ -2,8 +2,12 @@
 #
 # See ./contrib/utilities/run_clang_tidy.sh for details
 
-# We disable performance-inefficient-string-concatenation because we don't care about "a"+to_string(5)+...
+# Rationale for disabling warnings:
+#
+# - modernize-concat-nested-namespaces: This is c++17 only.
+# - performance-inefficient-string-concatenation: We don't care about "a"+to_string(5)+...
+# - performance-no-automatic-move: All modern compiler perform the return value optimization and we prefer to keep things const.
```

Comment:
There is also another case detected by `-Wretunrn-std-move` (http://clang.llvm.org/extra/clang-tidy/checks/performance-no-automatic-move.html) but since we are using `-Wall` with `clang` we should see that anyway.

Reply:
Yes. But we are not compiling with clang 10 in CI at the moment. That would be a good thing to do as well...